lib/v2/screens/dashboard/wallet_header.dart

```diff
@@ -19,74 +26,77 @@ class WalletHeader extends StatefulWidget {
 class WalletHeaderState extends State<WalletHeader> {
   final CarouselController _controller = CarouselController();
   int _selectedIndex = 0;
-
+  late TokenBalancesBloc _bloc;
   ScrollController dd = ScrollController();
 
+  @override
+  void initState() {
+    super.initState();
+    _bloc = TokenBalancesBloc()..add(const OnLoadTokenBalances());
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _bloc.close();
+  }
+
   void onPageChange(int index, CarouselPageChangedReason changeReason) {
     setState(() {
       _selectedIndex = index;
     });
   }
 
+  void reload() {
+    _bloc.add(const OnLoadTokenBalances());
+  }
+
   @override
   Widget build(BuildContext context) {
-    return Column(
-      children: <Widget>[
-        SingleChildScrollView(
-          child: CarouselSlider(
-            carouselController: _controller,
-            items: [
-              const Padding(
-                padding: EdgeInsets.only(right: 10),
-                child: CurrencyInfoCardWidget(
-                  token: SeedsToken,
-                  backgroundImage: 'assets/images/wallet/currency_info_cards/seeds/background.jpg',
-                  logo: 'assets/images/wallet/currency_info_cards/seeds/logo.jpg',
-                  balanceSubTitle: 'Wallet Balance 1',
+    return BlocBuilder<TokenBalancesBloc, TokenBalancesState>(
+        bloc: _bloc,
```

Comment:
Also with BlocProvider you avoid use the `bloc: _bloc`

Reply:
Same..